Modules/_json.c

```diff
@@ -317,18 +316,22 @@ static void
 raise_errmsg(const char *msg, PyObject *s, Py_ssize_t end)
 {
     /* Use JSONDecodeError exception to raise a nice looking ValueError subclass */
-    static PyObject *JSONDecodeError = NULL;
-    PyObject *exc;
+    _Py_static_string(PyId_decoder, "json.decoder");
```

Comment:
Do we really need PyId_decoder and PyId_JSONDecodeError micro-optimization? Decoding an ASCII string is fast in Python. Why not keeping PyImport_ImportModule & PyObject_GetAttrString?

I don't care much about the performance of the error path.

Reply:
I think json may be a little special - many people use it to load from sources that may provide invalid json data, this is pretty common:
```python
try:
    val = json.loads(...)
except json.JSONDecodeError:
    pass
else:
    ...
```
So raising error is quite a common operation. Using PyId here surprisingly seems to speed error raising by 20%.